### PR TITLE
feat: initialized UI for QR screen + installed RNCamera for Android

### PR DESCRIPTION
**Pull request by SpQuyt**

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
     <uses-permission android:name="android.permission.INTERNET" />
+    <!-- Required -->
+    <uses-permission android:name="android.permission.CAMERA" />
+
+    <!-- Include this only if you are planning to use the camera roll -->
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+
+    <!-- Include this only if you are planning to use the microphone for video recording -->
+    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
```
Comment (namdaoduy) [inline android/app/src/main/AndroidManifest.xml]: Remove this

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -3,6 +3,15 @@
     package="com.ee2_mobile_app">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <!-- Required -->
+    <uses-permission android:name="android.permission.CAMERA" />
+
+    <!-- Include this only if you are planning to use the camera roll -->
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (namdaoduy) [inline android/app/src/main/AndroidManifest.xml]: Remove this

File: package.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
     "query-string": "^6.8.3",
     "react": "16.8.3",
     "react-native": "^0.59.9",
+    "react-native-camera": "^3.14.0",
```
Comment (namdaoduy) [inline package.json]: Remove all `^` character in every package version to lock current working version

File: src/components/Home/QR/ResultCheckin/index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  componentWillMount() {
+    const { navigation, checkQRDispatch } = this.props;
+    const eventId = navigation.getParam('eventId');
+    const barcodeResult = navigation.getParam('barcodeResult');
+    const uid = Auth.getUserId();
+    checkQRDispatch(barcodeResult, eventId, uid)
+      .then(res => {
+        if (!res.success) {
+          // Specific errors will be declared here
```
Comment (namdaoduy) [inline src/components/Home/QR/ResultCheckin/index.js]: Add TODO tag

File: src/components/Home/QR/index.js [hunk 5]
```diff
@@ -1,25 +1,47 @@
...
-    backgroundColor: '#F5FCFF',
-  },
-});
+import { connect } from 'react-redux';
+import { RNCamera } from 'react-native-camera';
+import { checkQRDispatch } from 'datalayer/actions/qr.action';
+import NavigationWithoutProps from 'utils/navigationWithoutProps';
+import styles from './index.styles';
 
 class QR extends Component {
```
Comment (namdaoduy) [inline src/components/Home/QR/index.js]: Check if camera will be unmounted after we leave this screen

File: src/datalayer/reducers/qr.reducer.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { qrAction } from 'constants/actions';
+
+// This state is for getting check-in QR result
+export const INITIAL_STATE = {
+  checkedInResult: {},
```
Comment (namdaoduy) [inline src/datalayer/reducers/qr.reducer.js]: `checkedInResult` or `list`?

File: src/datalayer/store/promiseMiddleware.js [hunk 7]
```diff
@@ -39,6 +39,7 @@ export default ({ dispatch, getState }) => (next) => (action) => {
       extraPayload: rest.payload,
     });
 
+    console.log(error);
```
Comment (namdaoduy) [inline src/datalayer/store/promiseMiddleware.js]: Please remove

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (SpQuyt) [reply android/app/src/main/AndroidManifest.xml]: ok

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (SpQuyt) [reply android/app/src/main/AndroidManifest.xml]: ok

File: package.json (hunk 3 above)
Comment (SpQuyt) [reply package.json]: ok

File: src/datalayer/store/promiseMiddleware.js (hunk 7 above)
Comment (SpQuyt) [reply src/datalayer/store/promiseMiddleware.js]: ok

File: src/components/Home/QR/index.js (hunk 5 above)
Comment (SpQuyt) [reply src/components/Home/QR/index.js]: Cu yen tam. Trust me

File: src/datalayer/reducers/qr.reducer.js (hunk 6 above)
Comment (SpQuyt) [reply src/datalayer/reducers/qr.reducer.js]: ***checkedInResult*** of course
***list*** is for the list of events get by userId

File: src/components/Home/QR/ResultCheckin/index.js (hunk 4 above)
Comment (SpQuyt) [reply src/components/Home/QR/ResultCheckin/index.js]: ok

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (namdaoduy) [reply android/app/src/main/AndroidManifest.xml]: I mean remove the unused permissions, not remove the comments. Please read the permissions before put all of them to the app.

File: src/components/Home/QR/ResultCheckin/index.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    const eventId = navigation.getParam('eventId');
+    const barcodeResult = navigation.getParam('barcodeResult');
+    const uid = Auth.getUserId();
+    checkQRDispatch(barcodeResult, eventId, uid)
+      .then(res => {
+        if (!res.success) {
+          /**
+           * @todo Specific errors will be declared here.
+           * @todo Depends on how many type of errors that API response.
+           */
```
Comment (namdaoduy) [inline src/components/Home/QR/ResultCheckin/index.js]: This is OK but `// TODO: something` is enough, with single line comment. New IDE like VSCODE can help tracking TODOs like that.

File: src/datalayer/reducers/qr.reducer.js (hunk 6 above)
Comment (namdaoduy) [reply src/datalayer/reducers/qr.reducer.js]: I can not see any action that change `checkedInResult`. And you haven't set initial value for `list`.

File: src/components/Home/QR/ResultCheckin/index.js (hunk 8 above)
Comment (SpQuyt) [reply src/components/Home/QR/ResultCheckin/index.js]: sorry my bad

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (SpQuyt) [reply android/app/src/main/AndroidManifest.xml]: Okay, I have removed RECORD_AUDIO permission.

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (namdaoduy) [reply android/app/src/main/AndroidManifest.xml]: Consider removing these permission too. QR code scanning may not use camera roll permissions.

File: src/datalayer/reducers/qr.reducer.js (hunk 6 above)
Comment (SpQuyt) [reply src/datalayer/reducers/qr.reducer.js]: We will use it when we display the information of ticket. I just leave the success flow as ***Thành công*** not implemented ***checkedInResult*** yet.

Comment (SpQuyt) [reply src/datalayer/reducers/qr.reducer.js]: Oh yeah I forgot the ***list***. It has to be ***checkedInResult***

File: src/datalayer/reducers/event.reducer.js [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
 export const INITIAL_STATE = {
-  list: [],
+  eventList: [],
   eventDetail: {},
 };
 
 export default (state = INITIAL_STATE, action) => {
   switch (action.type) {
-    case eventAction.LOAD_EVENTS_LIST_SUCCESS: {
+    case eventAction.LOAD_EVENTS_eventList_SUCCESS: {
```
Comment (namdaoduy) [inline src/datalayer/reducers/event.reducer.js]: `LOAD_EVENTS_eventList_SUCCESS` <- be careful when using replace all

Comment (SpQuyt) [reply src/datalayer/reducers/event.reducer.js]: Oke

**Pull request merged**